packages/loader/container-loader/src/containerContext.ts

```diff
@@ -270,18 +288,43 @@ export class ContainerContext implements IContainerContext {
         return this.container.getAbsoluteUrl(relativeUrl);
     }
 
-    private async load() {
-        if (this.codeDetails === undefined) {
-            const nullChaincode =  new NullChaincode();
-            this._runtime = await nullChaincode.instantiateRuntime(this);
-            return;
+    /**
+     * Determines if the current code details of the context
+     * satisfy the incoming constraint code details
+     */
+    public async satisfies(constraitCodeDetails: IFluidCodeDetails) {
```

Comment:
Why is this logic in ContainerContext rather than Container?

Reply:
the context owns the runtime (fluidExport/runtime factory), which i think is the right division, so this must be here if the runtime is accessed.

Reply:
Maybe I'm mistaken, but I think of ContainerContext as less functionality and more data/proxy between Container and ContainerRuntime.

The only thing it depends on within ContainerContext is `this.fluidModuleP` which is only because code loading has moved from Container to ContainerContext already.

I don't think it's a rule we have to live by, but it was my understanding at least.  One benefit of it being this way is that the fluidModule lives and dies with the ContainerRuntime, which seems good.